src/service/history-impl.js

```diff
@@ -751,9 +751,15 @@ export class HistoryBindingVirtual_ {
   push() {
     // Current implementation doesn't wait for response from viewer.
     this.updateStackIndex_(this.stackIndex_ + 1);
+    // TODO(dvoytenko, #8785): cleanup after tracing.
```

Comment:
Then we'd loose the initial trace, i.e. who and how called this method. That's basically the problem I'm trying to address here. To preserve the original trace stack, I have to create error in original context, i.e. before I go into microtasks.